### ISSUE TITLE
Partial views do not get sync'd

### ISSUE THREAD

**Issue by UnsungHeros** [KevinJump/uSync8#107]
**Describe the bug**
When importing templates, views which are located in the partial views section do not get brought over.

**To Reproduce**
1. Create a view under partial views
2. Do an export
3. Check the usync templates folder

**Expected behavior**
Any partial view should show up.  Maybe not so much in the templates folder, but maybe a partial templates folder.

**Screenshots**

**Desktop (please complete the following information):**
 - OS: ALL
 - Browser [e.g. chrome, safari]:Chrome
 - Version: Latest

**Additional context**
As far as getting around this, we just manually copy up the files to our staging/prod environment.  Then we go in and add the folder/files that are needed.

**Comment by KevinJump** [comment]
Hi, 

this is as expected behaviour, uSync doesn't copy the physical files, (templates, scripts or css) only the elements that are stored in the database - You need to copy them elements over as part of your deployment.  (so that's usally at least `views,css,scripts`). 

_unless this is a usync.complete issue ? (that does push these things between servers as part of a publish)_

**Comment by UnsungHeros** [comment]
Hi Kevin:

I understand that USync doesn’t copy physical files.  I was referring to the listing in the back office.  Here is what hopefully illustrates the issue a little better:

--Kept my code the same and simply restored the database from before I ran usync
--Verified there were no doctypes, templates, etc in the back office
--Ran usync
--Reloaded the document types in the back office and they were there
--Reloaded the templates in the back office, and they were there
--Reload the partial views folder in the back office, and this is what was missing(the database entries the back office reads).

**Comment by KevinJump** [comment]
Hi I might be confused about what you are looking at 😕 but the partial views folder inside Umbraco is a listing of the `Views/Partials` folder from disk, there are no database entries for this. I would check that folder is present on your site.

**Comment by UnsungHeros** [comment]
The database entries I refer to are the database entries that are used by the back office to display the templating section entries in the back office.

One can physically create a file in the views folder, but Umbraco will not see the view until it is explicitly created in the back office because there is no database entry for it yet.  That is the concept I am talking about.  The same concept applies to the partial views folder IN the back office.  Those are the files that are missing.  Usync gets the templates(Back Office) /Views Filesystem), but it does not get the Partial Views(Back Office) /Views/Partials(Filesystem).

If you still cannot understand, go ahead and close the bug, because I don’t think I have any better alternative explanations of the issue.

**Comment by KevinJump** [comment]
Hi, I really don't want to close the issue if you think there is a problem 😦  If there is something we can do to fix it we will. 

From my understanding what we are saying is: 

## Templates in the back office 

![image](https://user-images.githubusercontent.com/431231/82907595-b889c400-9f5e-11ea-9788-0104800be1ef.png)

**This is working ?** 
This is the files in the root of the /views folder and as you say they do have a database entry for these items it contains the id (used in doc-types etc) and the hierarchy (so parent ids) - if the files are present in the views folder a uSync import will create the database entries for this section.

## Partial Views 

![image](https://user-images.githubusercontent.com/431231/82907738-ea028f80-9f5e-11ea-848e-f9ff10589642.png)

**this isn't working ?** 
There is no database entries for this bit ☹️ Internally Umbraco uses a [PartialViewRepository ](https://github.com/umbraco/Umbraco-CMS/blob/v8/dev/src/Umbraco.Core/Persistence/Repositories/Implement/PartialViewRepository.cs) which you can see from the code is just using the file system underneath to get a list of items.

---
uSync doesn't copy the views folder at all, as this is part of the disk, its assumed it is copied along with the usync folder before an import is ran. 

Things that can go wrong with ( the template bit) is if the files are missing the import of the templates should fail. ([code here looks for the file on disk](https://github.com/KevinJump/uSync8/blob/v8/dev/uSync8.Core/Serialization/Serializers/TemplateSerializer.cs#L43), If its missing we report it as an error). 

I suppose file permissions could cause the partials not to list ? if the account running the site can't see the folder it might report it as blank back in the repository?

**Comment by UnsungHeros** [comment]
Okay.  I gave it one last shot.  I took the database from before I last synced it and put it on a developer box.  I then got the latest source that we had checked into visual studio and verified that none of the document types and templates were present.  I ran usync and got everything expected this time including the partial views. Since everything was Fresh, I am going to assume it must have been some kind of security issue on the branch I was using.  Looks like the issue is fixed.  Thank you for your help!
